server/src/repositories/asset.repository.ts

```diff
@@ -201,6 +201,10 @@ export class AssetRepository implements IAssetRepository {
     );
   }
 
+  async setReadOnlyForLibrary(libraryId: string, isReadOnly: boolean): Promise<void> {
+    await this.repository.update({ library: { id: libraryId } }, { isReadOnly });
```

Comment:
External libraries can have isVisible:false motion parts of extracted live photos and they need to be excluded from this update.

Reply:
okay. I've updated it, but you should check that I understood correctly: https://github.com/immich-app/immich/pull/9188/commits/3e19aef1f8f05adff6c8fbc248dc85ab1384ad77

Reply:
Yeah, I think that looks good. Will have to test later with motion photos to make sure are working as expected.